extensions/moderation/in_voice_moderation.py

```diff
@@ -0,0 +1,203 @@
+            deaf = discord.ui.Section(
+                discord.ui.TextDisplay(content=user.display_name), accessory=Deaf(user, ctx=channel_ctx)
+            )
+            container.add_item(item=mute)
+            container.add_item(item=deaf)
+            container.add_separator()
+        self.add_item(container)
+
+
+class DnDManager(commands.Cog):
+    def __init__(self, client):
+        self.client: Bot = client
+        self.logger = CustomLogger(self.qualified_name, self.client.boot_time)
+        # {
+        #   channel_id: user
+        # }
+        self.claimed = {}
+        self.requested_message = {}
+
+    @commands.slash_command(name="claim", description="Claim this channel to moderate it")
+    async def claim(self, ctx: discord.ApplicationContext):
+        if isinstance(ctx.channel, discord.VoiceChannel):
+            try:
+                guild_claim = self.claimed[ctx.channel_id]
+                await ctx.response.send_message(
+                    f"This channel is already claimed by {guild_claim.display_name}", ephemeral=True, delete_after=5.0
+                )
+            except KeyError:
+                self.claimed[ctx.channel_id] = ctx.user
+                await ctx.response.send_message("You claimed this channel", ephemeral=True, delete_after=5.0)
+        else:
+            await ctx.response.send_message("This is not a voice channel", ephemeral=True, delete_after=5.0)
+
+    @commands.slash_command(name="unclaim", description="Unclaims the channel")
+    async def unclaim(self, ctx: discord.ApplicationContext):
+        if isinstance(ctx.channel, discord.VoiceChannel):
+            try:
+                guild_claim = self.claimed[ctx.channel_id]
+                if guild_claim == ctx.user or ctx.user.guild_permissions.administrator:
+                    del self.claimed[ctx.channel_id]
+                    try:
+                        await self.requested_message[ctx.channel_id].delete_original_message()
+                        del self.requested_message[ctx.channel_id]
+                    except KeyError:
+                        pass
+                    for member in ctx.channel.members:
+                        if member.bot:
+                            continue
+                        await member.edit(deafen=False, mute=False)
+                    await ctx.response.send_message("You unclaimed this channel", ephemeral=True, delete_after=5.0)
+                else:
+                    await ctx.response.send_message(
+                        "Only the owner or an admin can unclaim the channel", ephemeral=True, delete_after=5.0
+                    )
+            except KeyError:
+                await ctx.response.send_message("Channel is unclaimed", ephemeral=True, delete_after=5.0)
+        else:
+            await ctx.response.send_message("This is not a voice channel", ephemeral=True, delete_after=5.0)
+
+    @commands.slash_command(name="moderate", description="moderate user in this channel")
```

Comment:
should we move it to a subcommand like `/mod voice moderate` ?!
maybe enhance description